src/AntiPlagiarism/AntiPlagiarism.Core/Method/MethodIndex.cs

```diff
@@ -26,5 +28,27 @@ public MethodIndex(string name, string path, string sourceCode, int line, int ch
 			Line = line;
             Character = character;
         }
-    }
+
+		public override bool Equals(object obj) => Equals(obj as MethodIndex);
+
+		public bool Equals(MethodIndex other) =>
+			Name == other?.Name && Path == other.Path && SourceCode == other.SourceCode && 
```

Comment:
What if `other` isnull?

Reply:
Name == other?.Name guards against that

Reply:
Agree.
Another thing: I think we need to compare using ordinal logic and do not rely on a user's current culture during strings comparison

Reply:
the string overrides "==" and "!=" operators, and they use ordinal comparison